go-controller/pkg/ovn/egressip.go

```diff
@@ -1626,6 +1636,17 @@ func (e *EgressIPController) syncStaleEgressReroutePolicy(cache egressIPCache) e
 					_, ok := cache.egressIPNameToPods[extractedEgressIPName]
 					return !ok
 				}
+				allNetworkNodeRedirectCache, ok := cache.egressNodeRedirectsCache.cache[networkName]
+				if !ok || len(allNetworkNodeRedirectCache) == 0 {
+					klog.Infof("syncStaleEgressReroutePolicy found invalid logical router policy (UUID: %s) because no assigned Nodes for EgressIP %s", item.UUID, eipName)
+					return true
+				}
+				networkNodeRedirectCache := limitToValidEgressNodes(eipName, allNetworkNodeRedirectCache)
+				if len(networkNodeRedirectCache) == 0 {
+					klog.Errorf("syncStaleEgressReroutePolicy could not find at least one node in the redirects "+
+						"cache after filtering for egress ip %q, UUID %q, programming error, removing logical router policy", eipName, item.UUID)
+					return true
+				}
```

Comment:
I think this would be better if it was just something like
```
networkNodeRedirectCache := limitToValidEgressNodes(eipName, cache.egressNodeRedirectsCache.cache[networkName])
if len(allNetworkNodeRedirectCache) == 0 {
    klog.Infof("syncStaleEgressReroutePolicy deleting invalid logical router policy %q because there are no existing nodes assigned to its EgressIP %s", item.UUID, eipName)
    return true
}
```